packages/history-utility/src/__tests__/history-utility.react.spec.tsx

```diff
@@ -319,4 +319,74 @@ describe('proxyWithHistory: react', () => {
       );
     });
   });
+
+  describe('edge cases', () => {
+    it('should update history when a state property is set to undefined', async () => {
+      const initialState: { count: number; foo?: string } = {
+        count: 0,
+        foo: 'bar',
+      };
+
+      const state = proxyWithHistory(initialState);
+
+      function handleIncrement() {
+        if (state.value.count === 1) {
+          state.value.foo = undefined;
+        } else {
+          state.value.foo = 'bar';
+        }
+        state.value.count++;
```

Comment:
I'm not certain what this test is trying to prove.. all changes within the same execution scope will cause a single history change as far as I know. This includes increasing the counter.

Can you provide a simpler test that sets a value from defined to undefined.

You can also use the `delete` keyword to remove a property on the object as well

I would be interested to see both scenarios.

Ensure to add an await before the expectation  is confirmed to simulate a change in execution context.

Reply:
I simplified the test case and I added another test case to check the behavior with `delete`.